test/dynamo/test_autograd_function.py

```diff
@@ -0,0 +1,255 @@
+# Owner(s): ["module: dynamo"]
+
+import math
+
+import torch
+
+import torch._dynamo.test_case
+import torch._dynamo.testing
+
+
+class CustomFunc1(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, foo):
+        return foo + foo
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        return grad_output
+
+
+class CustomFunc2(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, foo):
+        return foo + foo
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        return grad_output
+
+
+class CustomFunc3(torch.autograd.Function):
+    # Test there is graph break in forward function
+    @staticmethod
+    def forward(ctx, foo):
+        result = foo + foo
+        torch._dynamo.graph_break()
+        result = result + foo
+        ctx.save_for_backward(result)
+        return result
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        (result,) = ctx.saved_tensors
+        return grad_output * math.sqrt(result.numel())
+
+
+class Module1(torch.nn.Module):
+    def forward(self, foo):
+        return CustomFunc1().apply(foo)
+
+
+class Module2(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.fn = CustomFunc1.apply
+
+    def forward(self, foo):
+        return self.fn(foo)
+
+
+class Module3(torch.nn.Module):
+    def forward(self, foo):
+        return CustomFunc2().apply(foo)
+
+
+class Module4(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.fn = CustomFunc2.apply
+
+    def forward(self, foo):
+        return self.fn(foo)
+
```

Comment:
Modules 1-4 seem unused?

Also, we have almost no small tests that the mechanism actually works (aside from the graph break counts in the CSV and test_repros). We should add some sanity checks for some simple autograd.Function that Dynamo doesn't graph break

Reply:
Accessed in a loop test via `model = globals()[f"Module{i}"]()`